source/common/ssl/context_impl.cc

```diff
@@ -209,7 +209,7 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
         ASN1_STRING* str = altname->d.uniformResourceIdentifier;
         char* crt_san = reinterpret_cast<char*>(ASN1_STRING_data(str));
```

Comment:
a side question though, if we use `string (const char* s, size_t n)` then isn't it also O(n)?